main/realmain.cc

```diff
@@ -715,6 +715,18 @@ int realmain(int argc, char *argv[]) {
                 auto file = gs->enterFile(string("-e"), input);
                 inputFiles.emplace_back(file);
             }
+            if (!opts.inlineRBIInput.empty()) {
+                prodCounterAdd("types.input.bytes", opts.inlineRBIInput.size());
+                prodCounterInc("types.input.lines");
+                prodCounterInc("types.input.files");
+                auto input = opts.inlineRBIInput;
+                if (core::File::fileStrictSigil(opts.inlineRBIInput) == core::StrictLevel::None) {
+                    // put it at the end so as to not upset line numbers
+                    input += "\n# typed: true";
+                }
+                auto file = gs->enterFile(string("--rbi.rbi"), input);
+                inputFiles.emplace_back(file);
+            }
```

Comment:
Let's factor this to a helper function, and call it once for `-e` and once for `--rbi`.